src/api/schema/components/source.rs

```diff
@@ -69,10 +72,9 @@ impl sort::SortableByField<SourcesSortFieldName> for Source {
             SourcesSortFieldName::ComponentType => {
                 Ord::cmp(self.get_component_type(), rhs.get_component_type())
             }
-            SourcesSortFieldName::OutputType => Ord::cmp(
-                &SourceOutputType::from(self.0.output_type),
-                &SourceOutputType::from(rhs.0.output_type),
-            ),
+            SourcesSortFieldName::OutputType => {
+                Ord::cmp(&u8::from(self.0.output_type), &u8::from(rhs.0.output_type))
```

Comment:
👍🏻 It might be worth adding a comment to `SourceOutputType` to keep its options alphabetical (which they already are), so that bitmask values match the intuitive sort order users likely expect.

Just in case this gets added to in the future and that connection isn't obvious.

Reply:
Indeed, the fact that the enum order drives how sorting happens definitely needs to be documented.